bytewriter/bytewriter.go

```diff
@@ -4,14 +4,15 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	mmap "github.com/edsrzf/mmap-go"
 )
 
 // assumes Little Endian, use _arch.go to set it to BigEndian for those archs
 var byteOrder = binary.LittleEndian
 
 // ByteWriter is a simple wrapper over a byte slice that supports writing anywhere
 type ByteWriter struct {
-	buffer []byte
+	buffer mmap.MMap
```

Comment:
This is not needed. `mmap.MMap` is a byte slice. Doing `[]byte(mapping)` will get you the `[]byte` object